usr/src/grub/grub-0.97/stage2/fsys_zfs.c

```diff
@@ -1054,6 +1054,8 @@ static const char *spa_feature_names[] = {
 	"com.delphix:embedded_data",
 	"org.open-zfs:large_blocks",
 	"org.illumos:sha512",
+	"com.delphix:redaction_bookmarks",
```

Comment:
Why need to have 2 feature flags? just for easy tracking?

Reply:
It's primarily for tracking purposes. By having a featureflag for each one, we can easily track whether each feature is active or enabled, which allows pools that have had redaction bookmarks (or redacted datasets) added and then deleted to be read by older systems.  The two features are also fundamentally differently implemented, so it makes sense for them to be tracked separately even if they're introduced by the same code to support the same workflow.

Reply:
The `redaction_bookmarks` would be active on the sending pool, and `redacted_datasets` on the receiving pool.  So you'd typically have only one of these features active on a given pool (unless the pool is both a sender and a receiver).  This isn't a super strong argument for having different feature flags, but hopefully it helps explain why we think about them differently.